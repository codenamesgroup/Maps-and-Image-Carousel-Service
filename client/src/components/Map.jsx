import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { GOOGLE_API_KEY } from './../keys';

class Map extends React.PureComponent {
	static propTypes = {
		longitude: PropTypes.number,
		latitude: PropTypes.number,
	}

	static defaultProps = {
		longitude: 0.0,
		latitude: 0.0,
	}

	static miniMap = styled.img`
		border: 1px solid #cccccc;
		width: 285px;
		height: 135px;
		cursor: pointer;
	`;

	onClick = () => {
		console.log(this.props.latitude, this.props.longitude);
	}

	getMapUri = () => {
		return `
			https://maps.googleapis.com/maps/api/staticmap
				?key=${GOOGLE_API_KEY}
				&zoom=13
				&size=285x135
				&maptype=roadmap
				&markers=color:red%7Clabel:S%7C
				${this.props.latitude},
				${this.props.longitude}
		`;
	}

	render() {
		return (
			<Map.miniMap
				src={this.getMapUri()}
				onClick={this.onClick}
			/>
		);
	}
}

export { Map };