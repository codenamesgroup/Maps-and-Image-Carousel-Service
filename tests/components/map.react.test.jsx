import React from 'react';
import { shallowToJson } from 'enzyme-to-json';

import { Map } from './../../client/src/components/Map.jsx';

describe('<Map/>', () => {

	it('should render with long and lat props', () => {
		let component = shallow(<Map longitude={0} latitude={0}/>);
		expect(shallowToJson(component)).toMatchSnapshot();
	});

	it('should render without props', () => {
		let component = shallow(<Map/>);
		expect(shallowToJson(component)).toMatchSnapshot();
	});
});