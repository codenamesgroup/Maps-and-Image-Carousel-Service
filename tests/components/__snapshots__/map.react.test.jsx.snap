// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map/> should render with long and lat props 1`] = `
<styled.img
  onClick={[Function]}
  src="
			https://maps.googleapis.com/maps/api/staticmap
				?key=AIzaSyAu2ghVBWddafMcKs6b__fCF7hGkTyb0v8
				&zoom=13
				&size=285x135
				&maptype=roadmap
				&markers=color:red%7Clabel:S%7C
				0,
				0
		"
/>
`;

exports[`<Map/> should render without props 1`] = `
<styled.img
  onClick={[Function]}
  src="
			https://maps.googleapis.com/maps/api/staticmap
				?key=AIzaSyAu2ghVBWddafMcKs6b__fCF7hGkTyb0v8
				&zoom=13
				&size=285x135
				&maptype=roadmap
				&markers=color:red%7Clabel:S%7C
				0,
				0
		"
/>
`;
